polymath/extensions/pickler.py

```
@@ -90,920 +90,930 @@
+        else:
+            digits = None
 
-    if hasattr(self, '_pickle_reference'):
-        reference = self._pickle_reference
-    else:
-        reference = None
+        self._pickle_digits = QubePickler._validate_pickle_digits(digits, reference)
 
-    reference = _validate_pickle_reference(reference)
-    self._pickle_reference = reference
+        for key, deriv in self._derivs.items():
```

Ruff (0.14.7)

Loop control variable `key` not used within loop body

Rename unused `key` to `_key`

(B007)